cmd/controller.go

```diff
@@ -136,18 +123,24 @@ func newControllerCommand() *cobra.Command {
 func watchConfig(ctx context.Context, argocdService argocd.Service, clientset kubernetes.Interface, namespace string, callback func(map[string]triggers.Trigger, map[string]notifiers.Notifier, *settings.Config) error) {
 	var secret *v1.Secret
 	var configMap *v1.ConfigMap
+	var defaultConfig *settings.Config
 	lock := &sync.Mutex{}
-	onNewConfigMapAndSecret := func(newSecret *v1.Secret, newConfigMap *v1.ConfigMap) {
+	onNewConfigMapAndSecret := func(newSecret *v1.Secret, newConfigMap *v1.ConfigMap, newBuiltitCm *v1.ConfigMap) {
```

Comment:
Instead of introducing a separate config map for built-in template can we put "built-in" templates and triggers into existing default config map while changing the settings structure from: 

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: argocd-notifications-builtin-cm
data:
  config.yaml: |
    templates:
    - body: |
        {{if eq .context.notificationType "slack"}}:exclamation:{{end}} Application {{.app.metadata.name}} has degraded.
        Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
      name: app-health-degraded
      title: Application {{.app.metadata.name}} has degraded.
```


```yaml

apiVersion: v1
kind: ConfigMap
metadata:
  creationTimestamp: null
  name: argocd-notifications-builtin-cm
data:
  templates.app-health-degraded: |
    body:
      {{if eq .context.notificationType "slack"}}:exclamation:{{end}} Application {{.app.metadata.name}} has degraded.
      Application details: {{.context.argocdUrl}}/applications/{{.app.metadata.name}}.
    title: Application {{.app.metadata.name}} has degraded.
```

The idea is to make configuration "kustomize"/"kubectl apply" friendly so that user can get existing configuration and "insert" new templates, triggers and "override" existing using template/trigger name.

Reply:
See https://github.com/argoproj-labs/argocd-notifications/pull/122/files#diff-9dc9a1ee4349973b3e96fcbf3aa4eed184aaf68f3761f36548d125080249d660R3-R164 please, Is that what you have in mind?

Reply:
Just noticed that this should also be documented as a new feature.
Do we need also to add to the `config.yaml` deprecation notice?

Reply:
@alexmt  done. I think that calling it `trigger.NAME` and `template.NAME` instead of `triggers` and `templates` will be less error prune, wdyt?

Reply:
@olegsu I think you are right. Agree that it looks better.